app/models/slot_constraints.rb

```diff
@@ -0,0 +1,35 @@
+class SlotConstraints
+  delegate :map, to: :@slot_availabilities
+
+  def initialize(slot_availabilities)
+    @slot_availabilities = slot_availabilities
+  end
+
+  def bookable_date?(requested_date)
+    @slot_availabilities.any? do |slot_availability|
+      slot_availability.slot.to_date == requested_date &&
+        slot_availability.bookable?
+    end
+  end
+
+  def bookable_slot?(requested_slot)
+    @slot_availabilities.any? do |slot_availability|
+      slot_availability.slot == requested_slot &&
+        slot_availability.bookable?
+    end
+  end
+
+  def last_bookable_date
+    bookable_slots.sort.last.slot.to_date
```

Comment:
I think we could get this to be more concise, something like:

```ruby
bookable_slots.max.to_date
```

`max` instead of `sort.last` and delegating `to_date` to slot in `SlotAvailability`